### PR TITLE
Remove valid score and cleanup

### PR DESCRIPTION
**Pull request by MemeHovy**
removed valid score since it seemed useless

### CODE REVIEW

File: source/Highscore.hx [hunk 1]
```diff
@@ -7,19 +7,11 @@ using StringTools;
 */
 class Highscore
 {
-	#if (haxe >= "4.0.0")
```
Comment (YanniZ06) [inline source/Highscore.hx:10]: Good, people should update anyways

File: source/PlayState.hx [hunk 2]
```diff
@@ -5997,16 +5997,11 @@ class PlayState extends MusicBeatState
...
-			{
-				#if !switch
-				var percent:Float = ratingPercent;
-				var letter:String = ratingName;
-				var intensity:String = ratingIntensity;
-				if(Math.isNaN(percent)) percent = 0;
-				Highscore.saveScore(SONG.header.song, songScore, storyDifficulty, percent, ratingName, ratingIntensity);
-				#end
-			}
+			var percent:Float = ratingPercent;
```
Comment (YanniZ06) [inline source/PlayState.hx:6000]: Add back switch check, I’m not certain about what it does but it shouldn’t just be removed

**Review by YanniZ06 — CHANGES_REQUESTED** [review]
Solid i’d say, if you add back the switch conditional flag it should be all good!

File: source/Song.hx [hunk 3]
```diff
@@ -203,11 +202,15 @@ class Song
...
 		try {
 			songJson = cast Json.parse(rawJson).song;
-			songJson.header.validScore = true;
 		} catch (e) {
 			//yes honey i know you work, now please shut the fuck up <3
 			//trace('\n<----ERROR---->\n' + e + '\nSong JSON was detected invalid/old, attempting conversion...\n<-------->');
 			oldSongJson = cast Json.parse(rawJson).song;
+			for (e in oldSongJson) {
+				if (e.validScore != null){
+					Reflect.deleteField(e, "validScore");
```
Comment (UmbratheUmbreon) [inline source/Song.hx:211]: Dont need to delete the field, it will be dropped anyways since the typedef does not contain it.

**Review by UmbratheUmbreon — CHANGES_REQUESTED** [review]
It would be fine excluding the one thing, and what yanni mentioned.

File: source/Song.hx (hunk 3 above)
Comment (MemeHovy) [reply source/Song.hx:211]: I removed it

**Comment by MemeHovy** [comment]
I’m just gonna close this since I already did it in the dev repo @UmbratheUmbreon and @YanniZ06

**Pull request closed without merging**